Tests/L0/SonarQubePostTest/DisableAnalysisOnPrBuild.ps1

```diff
@@ -4,13 +4,68 @@ param()
 . $PSScriptRoot\..\..\lib\Initialize-Test.ps1
 
 # Arrange 
-
 $distributedTaskContext = 'Some distributed task context'
-Register-Mock Get-TaskVariable { $true } -- -Context $distributedTaskContext -Name 'DisableSQAnalysisOnPrBuilds'
-Register-Mock IsPrBuild {$true}
-Register-Mock Write-Host
 
-# Act - this script contains executable code and would fail if allowed to run 
-. $PSScriptRoot\..\..\..\Tasks\SonarQubePostTest\SonarQubePostTest.ps1 -connectedServiceName "service" -projectKey "projectKey" -projectName "projectName" -projectVersion "1"
+function VerifyPrRun
+{
+    param([string]$SQPullRequestBotSetting, [bool]$IsPrBuild, [bool]$ExpectedToRun )
+
+    # Arrange
+    if ($SQPullRequestBotSetting -ne $null)
+    {
+        Register-Mock Get-TaskVariable { $SQPullRequestBotSetting } -- -Context $distributedTaskContext -Name 'SQPullRequestBot'
+    }
+    
+    Register-Mock IsPrBuild {$IsPrBuild}
+    Register-Mock Write-Host
+    Register-Mock InvokeMSBuildRunnerPostTest
+    Register-Mock UploadSummaryMdReport
+    Register-Mock HandleCodeAnalysisReporting
+    Register-Mock BreakBuildOnQualityGateFailure
+    
+    #Act
+    . $PSScriptRoot\..\..\..\Tasks\SonarQubePostTest\SonarQubePostTest.ps1
+    
+    # Assert
+    if ($ExpectedToRun)
+    {
+        Assert-WasCalled InvokeMSBuildRunnerPostTest
+    }
+    else
+    {
+        Assert-WasCalled Write-Host -ArgumentsEvaluator {$args[0] -like "*SQPullRequestBot*"}
+        Assert-WasCalled InvokeMSBuildRunnerPostTest -Times 0
+    }
+    
+    # Cleanup
+    Unregister-Mock IsPrBuild 
```

Comment:
Out of interest, how do the _Assert_ methods work? Presumably they don't halt execution otherwise the clean-up code won't be hit if an assertion fails.